### PR TITLE
NMS-13739: Add OIA plugin support for Minion

### PR DESCRIPTION
**Pull request by j-white**
JIRA: https://issues.opennms.org/browse/NMS-13739

Refactor the API layer and allow parts of the extensions to be installed on Minion.

Also managed to reproduce and resolve https://issues.opennms.org/browse/NMS-13787 during this effort.

### CODE REVIEW

File: container/features/pom.xml [hunk 1]
```diff
@@ -85,9 +85,64 @@
...
+                          <descriptor>file:${project.build.directory}/classes/features-minion.xml</descriptor>
+                          <descriptor>file:${project.build.directory}/classes/spring.xml</descriptor>
+                      </descriptors>
+                      <distribution>org.apache.karaf.features:framework</distribution>
+                      <framework>
+                          <feature>framework</feature>
+                          <feature>aries-blueprint</feature>
+                      </framework>
+                      <features>
+                          <feature>minion-api-layer</feature>
```
Comment (j-white) [inline container/features/pom.xml:118]: This plugin makes it possible to test that features are installable and contain all the required bundles. Works reliably too!

File: smoke-test/src/test/java/org/opennms/smoketest/IntegrationAPIIT.java [hunk 2]
```diff
@@ -48,21 +48,36 @@
...
+        }
+
+        // Now wait until the health check passes on OpenNMS
+        verifyHealthCheckWithDescription(opennmsSsh, "Sample Project :: Health Check");
+
+        // Now do the same on Minion
+        verifyHealthCheckWithDescription(minionSsh, "Sample Project :: Minion");
+
+        // Now ensure that we're able to invoke the extensions on Minion from OpenNMS
+        verifyHealthCheckWithDescription(opennmsSsh, "Sample Project :: Service Extensions on Minion");
```
Comment (j-white) [inline smoke-test/src/test/java/org/opennms/smoketest/IntegrationAPIIT.java:80]: This verifies that the extensions can actually be invoked on Minion and tests the round-trip.

File: pom.xml [hunk 3]
```diff
@@ -1468,8 +1468,8 @@
     <powermockVersion>2.0.9</powermockVersion>
     <okhttpVersion>3.10.0</okhttpVersion>
     <okioVersion>1.14.0</okioVersion>
-    <opennmsApiVersion>0.6.0</opennmsApiVersion>
-    <opennmsApiVersionOsgi>0.6.0</opennmsApiVersionOsgi>
+    <opennmsApiVersion>0.7.0-SNAPSHOT</opennmsApiVersion>
```
Comment (j-white) [inline pom.xml:1471]: Updating OIA to 0.7.0-SNAPSHOT - we expect further updates before cutting a release.

**Review by cgorantla — APPROVED** [review]
Looks great.  Just to be safe, good to trigger full smoke test suite before merging.

**Comment by j-white** [comment]
Full smoke test passed. Merging.

**Pull request merged**